python/kwiver/sprokit/adapters/adapter_data_set.cxx

```diff
@@ -128,6 +131,36 @@ py::object get_port_data_correct_type(ka::adapter_data_set &self, ::sprokit::pro
   throw py::type_error(msg);
 }
 
+template < typename T >
+py::object value_or(ka::adapter_data_set &self,
+                    ::sprokit::process::port_t const& port,
+                    py::object value_if_missing=py::none())
+{
+  try
+  {
+    return py::cast(self.value<T>(port));
+  }
+  catch(std::runtime_error const& e)
+  {
+    return value_if_missing;
+  }
+}
+
+py::object value_or_correct_type(ka::adapter_data_set &self,
+                                 ::sprokit::process::port_t const& port,
+                                 py::object value_if_missing=py::none())
+{
+  try
+  {
+    return get_port_data_correct_type(self, port);
+  }
+  catch(std::runtime_error const& e)
+  {
+    return value_if_missing;
+  }
+}
+
```

Comment:
Can we avoid consecutive blank lines, please?

Reply:
Resolved in squashed commit a56c980c7929a6e271e10414de955725f8c0ba3f